torch/linalg/__init__.py

```diff
@@ -204,31 +213,42 @@
 slogdet = _add_docstr(_linalg.linalg_slogdet, r"""
 linalg.slogdet(input, *, out=None) -> (Tensor, Tensor)
 
-Calculates the sign and natural logarithm of the absolute value of a square matrix's determinant,
-or of the absolute values of the determinants of a batch of square matrices :attr:`input`.
-The determinant can be computed with ``sign * exp(logabsdet)``.
+Computes the sign and logarithm of the norm of the determinant of a square matrix or batch of square matrices.
 
-Supports input of float, double, cfloat and cdouble datatypes.
+For a complex :attr:`input`, as the determinant is complex in this case, it returns the angle
+and the logarithm of the modulus of the determinant.
 
-.. note:: When given inputs on a CUDA device, this function synchronizes that device with the CPU.
+The returend value decomposition is represented as a namedtuple `(sign, logabsdet)`.
+If :attr:`input` is a batch of matrices, then `sign`, `logabsdet` are also batched with the same
+batch dimensions. The determinant can be recovered as `sign * exp(logabsdet)`.
 
-.. note:: The determinant is computed using LU factorization. LAPACK's `getrf` is used for CPU inputs,
-          and MAGMA's `getrf` is used for CUDA inputs.
+Supports :attr:`input` of float, double, cfloat and cdouble dtypes.
+`logabsdet` will always be real-valued, even if :attr:`input` is complex.
+`sign` will have the same dtype as :attr:`input`.
 
-.. note:: For matrices that have zero determinant, this returns ``(0, -inf)``.
-          If :attr:`input` is batched then the entries in the result tensors corresponding to matrices with
-          the zero determinant have sign 0 and the natural logarithm of the absolute value of the determinant -inf.
+.. note:: When :attr:`input` is on a CUDA device, this function synchronizes that device with the CPU.
+
+.. note:: This function is computed using :func:`torch.lu`.
+
+.. note:: For matrices that have zero determinant, this returns `(0, -inf)`.
+          If :attr:`input` is batched then the entries in the result tensors corresponding
+          to matrices with the zero determinant have sign 0 and the natural logarithm of
+          the norm of the determinant `-inf`.
+
+.. seealso::
+
+        :func:`torch.linalg.det` computes the determinant of a square matrix
 
 Args:
-    input (Tensor): the input matrix of size :math:`(n, n)` or the batch of matrices of size :math:`(*, n, n)`
-                    where :math:`*` is one or more batch dimensions.
+    input (Tensor): tensor of size `(*, n, n)` where `*` is zero or more batch dimensions
+                    consisting of `(n, n)` matrices.
 
 Keyword args:
-    out (tuple, optional): tuple of two tensors to write the output to.
+    out (tuple, optional): output tuple of two tensors. Ignored if `None`. Default: `None`.
 
 Returns:
     A namedtuple (sign, logabsdet) containing the sign of the determinant and the natural logarithm
```

Comment:
"containing the sign and natural logarithm of the norm of an input matrix's determinant or the signs and natural logarithms of the norms of input matrices' determinants."

Reply:
I'm deleting this return for the reasons given here: https://github.com/pytorch/pytorch/pull/56265#discussion_r615922987